cron_scripts/inactive_idle_accounts.pl

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+use lib 'lib';
+
+use Getopt::Long;
+use ZCS::API;
+use Data::Dumper;
+use JSON;
+use File::Slurp;
+use POSIX qw(strftime);
+
+my $config_file = 'config/config.json';
+
+my %opts = (
+    last_login_days => 97,
+    new_status      => 'locked'
+);
+
+my $res =
+  GetOptions( \%opts, "help", "debug", "last_login_days=i", "new_status=s" );
+
+if ( $opts{help} ) {
+    print_usage();
+    exit 0;
+}
+elsif ( !$res ) {
+    print_usage();
+    exit 1;
+}
+
+my $config = JSON->new->utf8->relaxed->decode( scalar read_file $config_file);
+die "MAILSTOREHOST is not defined in config file"
+  if ( !defined( $config->{MAILSTOREHOST} ) || $config->{MAILSTOREHOST} eq "" );
+die "MAILSTOREUSER is not defined in config file"
+  if ( !defined( $config->{MAILSTOREUSER} ) || $config->{MAILSTOREUSER} eq "" );
+die "MAILSTOREPWD is not defined in config file"
+  if ( !defined( $config->{MAILSTOREPWD} ) || $config->{MAILSTOREPWD} eq "" );
+
+my $new_status      = $opts{new_status};
+my $last_login_days = $opts{last_login_days};
+my $last_login_timestamp =
+  strftime( '%Y%m%d000000',
+    localtime( time() - $last_login_days * ( 24 * 60 * 60 ) ) )
+  . "Z";
+
+my $soap_api = ZCS::API->new(
+    conf => {
+        SOAPURI => "https://"
+          . $config->{MAILSTOREHOST}
+          . ":7071/service/admin/soap",
+        SOAPUser => $config->{MAILSTOREUSER},
+        SOAPPass => $config->{MAILSTOREPWD},
+    },
+    debug => ( $opts{debug} ) ? 1 : 0
+);
+
+# search for 'active' accounts whose last login is earlier than specified days $last_login_days
+my @data;
+if ($soap_api) {
+    my %args = (
+        "query" =>
+"(&(zimbraLastLogonTimestamp<=$last_login_timestamp)(zimbraAccountStatus=active)(!(zimbraIsAdminAccount=TRUE)))",
+        "attrs" => "uid,zimbraId,zimbraAccountStatus,zimbraLastLogonTimestamp"
+    );
+
+    if ( my $searched_users = eval { $soap_api->searchdirectory(%args) } ) {
+        my $path = "/Envelope/Body/SearchDirectoryResponse";
+        my $i    = 0;
+        foreach my $acc ( $searched_users->dataof("$path/account") ) {
+            $i++;
+            my $msom = $searched_users->match("$path/[$i]")
+              or die("MATCH '$path/[$i]' failed\n");
+
+            push @data, $acc->attr;
+        }
+    }
+    else {
+        warn(   "DEBUG: SearchDirectoryRequest "
```

Comment:
nit1: remove 'DEBUG' perhaps replace with 'ERROR'?  We're producing this output even when not in debug mode.
nit2: warn takes a LIST as an argument so don't use '.' (dot|period) to concatenate data, instead call with multiple arguments separated by a ',' (comma)